tests/Persistence/SqlAllMappingsLookupTest.php

```diff
@@ -77,66 +76,49 @@ public function newSqlAllMappingsLookup(): SqlAllMappingsLookup {
 	public function testUnmodifiedMappings(): void {
 		$allMappings = $this->newSqlAllMappingsLookup()->getAllMappings();
 
-		// TODO: database is polluted by other tests
-		// $this->assertCount( 4, $allMappings );
-
-		$this->assertNull(
-			$this->getMappingsForEntityId( new ItemId( 'Q99001' ), $allMappings )
-		);
-		$this->assertEquals(
-			[
-				new Mapping( 'foo:foo', 'https://example.com/#foo1' )
-			],
-			$this->getMappingsForEntityId( new ItemId( 'Q99002' ), $allMappings )->asArray()
-		);
-		$this->assertEquals(
-			[
-				new Mapping( 'foo:foo', 'https://example.com/#foo1' ),
-				new Mapping( 'foo:bar', 'https://example.com/#bar1' ),
-				new Mapping( 'foo:baz', 'https://example.com/#baz1' )
-			],
-			$this->getMappingsForEntityId( new ItemId( 'Q99003' ), $allMappings )->asArray()
-		);
-
-		$this->assertNull(
-			$this->getMappingsForEntityId( new PropertyId( 'P99001' ), $allMappings )
-		);
-		$this->assertEquals(
-			[
-				new Mapping( 'foo:foo', 'https://example.com/#foo1' )
-			],
-			$this->getMappingsForEntityId( new PropertyId( 'P99002' ), $allMappings )->asArray()
-		);
 		$this->assertEquals(
 			[
-				new Mapping( 'foo:foo', 'https://example.com/#foo1' ),
-				new Mapping( 'foo:bar', 'https://example.com/#bar1' ),
-				new Mapping( 'foo:baz', 'https://example.com/#baz1' )
+				new MappingListAndId(
+					new ItemId( 'Q99002' ),
+					new MappingList( [
+						new Mapping( 'foo:foo', 'https://example.com/#foo1' )
+					] )
+				),
+				new MappingListAndId(
+					new ItemId( 'Q99003' ),
+					new MappingList( [
+						new Mapping( 'foo:foo', 'https://example.com/#foo1' ),
+						new Mapping( 'foo:bar', 'https://example.com/#bar1' ),
+						new Mapping( 'foo:baz', 'https://example.com/#baz1' )
+					] )
+				),
+				new MappingListAndId(
+					new PropertyId( 'P99002' ),
+					new MappingList( [
+						new Mapping( 'foo:foo', 'https://example.com/#foo1' )
+					] )
+				),
+				new MappingListAndId(
+					new PropertyId( 'P99003' ),
+					new MappingList( [
+						new Mapping( 'foo:foo', 'https://example.com/#foo1' ),
+						new Mapping( 'foo:bar', 'https://example.com/#bar1' ),
+						new Mapping( 'foo:baz', 'https://example.com/#baz1' )
+					] )
+				)
 			],
-			$this->getMappingsForEntityId( new PropertyId( 'P99003' ), $allMappings )->asArray()
+			$allMappings
```

Comment:
The two tests in this class are succeeding in isolation but failing when running all tests. Also one is failing when running all tests in this class. 

The deserialization function is getting JSON of an item itself, rather than our slot:

![image](https://user-images.githubusercontent.com/146040/181934159-b06bbc37-7c8e-44eb-853d-a4f0602a4059.png)

Perhaps there is a bug in the SQL that causes content from main slots to also be selected. I looked at the SQL and did not spot an issue. That does not mean there is none. (I also tried getting rid of the table aliases to improve readability but then MediaWiki complained about not finding the tables.)

Reply:
I'll have a look at this branch.

The aliases were necessary because the test environment uses table prefixes and I couldn't find a different way to handle that.